logic/cleaned.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-console */
+const validator = require('email-validator')
+const Sequelize = require('sequelize')
+
+const sequelize = new Sequelize('customers_DB', 'root', '123456789', {
+  host: 'localhost',
+  dialect: 'mysql'
+})
+
+
+// we are taking the customers object read from file and converted into csv
+// we then running it against our email validator npm package
+// if email value is good we push it to clean emails if value is not clean we push it to bad 
+// list then attempt to clean it for the user
+// when we finished cleaning we should push the final data back to the cleaned already and return to the controller who will push back to the model 
+// or we could push it to model directly
+
+const cleaned = (req, res, customers) => {
+  let cleanedCustomersList = []
+  let emailsNeedingRepairs = []
+  let cleanedCustomers = customers.map((customer) => {
+    // this will show the valid emails and push
+    // into cleanedcustomerslist
+    if (validator.validate(customer.Email)) {
+      cleanedCustomersList.push(customer)
+    }
+    // else if there is a bad email
+    // it will push it to bad email list so we can fix them
+    emailsNeedingRepairs.push(customer)
+  })
+
+  // res.send(cleanedCustomersList)
+  console.log('here\'s the good emails ...')
+  console.log(cleanedCustomersList)
+  console.log('here\'s a list of the bad emails ...')
+  console.log(emailsNeedingRepairs)
+
+  // this was my logic to fix the bad emails
+  
+  // emailsNeedingRepairs.map((badEmail)=>{
+  //     const specialChars = "~!#$%^&*()_+-={}<>,?/':;"
+  //     const trailingDots = "..."
+  //     const space = " "
+  //     let repairedEmails = []
+  //     specialChars.forEach(sChar => {
```

Comment:
This implementation won´t work how you expect.
This **forEach** loop is going to iterate over the following string **~!#$%^&*()_+-={}<>,?/':;**.

So imagine we have the following email:** inva~lid)em ail@email.com**

The first loop: **sChar** value will be ~ right?
- so it is going to check that ~ is in the email string.
- it is in the email so what it is going to do is. Remove the ~ from the email, so the new email will be **invalid)em ail@email.com**.
- The email is going to be saved in the repairedEmail array. (Where we can see the email is still invalid)

Ok, next loop in the iteration: **sChar** value now is !
- The code is going to check if ! is in the email, and it is not.
- So it goes to the next if and it finds there is a white spot.
- The next step is to remove the white spot, so the new email is going to be **invalid)email@email.com**.
- The email will be added to the repairedEmail array.

What is the value of the current **repairedEmail**: [´invalid)em ail@email.com´, ´invalid)email@email.com´

You see the pattern here? This code only assumes there is one invalid character in the email list.
Also it checks while looping over the invalid characters other things, like trailing dots, or spaces. 

A way you could correct this is by extracting checking for white spaces and dots outside of the forEach loop.

But this is not the ideal solution because the complexity of this block of code is O^2 which means it is exponential so for the more customers the longer it will check to validate the customers.